cpp/src/interop/to_arrow_device.cu

```diff
@@ -603,7 +603,7 @@ unique_device_array_t create_device_array(nanoarrow::UniqueArray&& out,
   });
   result->device_id          = rmm::get_current_cuda_device().value();
   result->device_type        = ARROW_DEVICE_CUDA;
-  result->sync_event         = private_data->sync_event;
+  result->sync_event         = &private_data->sync_event;
```

Comment:
So this seems wrong to me. There is no cuda-event API that requires the address of the event object other than the cudaEventCreate ones which has already been called. 
The address of the event object is part of the `private_data` which is released 2 lines below. This is safe only because the `private_data` is stored in the `result->array` but it does seem a poor pattern.
What is the reason the address of the event is needed over just the event object itself?

Reply:
Ok, got it. I thought it may be something like this. Thanks for the reference link.